backend/src/connectors/v2/authorisation/base.ts

```diff
@@ -68,15 +69,14 @@ export class BasicAuthorisationConnector {
     return true
   }
 
-  async hasAccessRequest(user: UserDoc, model: ModelDoc) {
-    const entities = await authentication.getEntities(user)
-
-    const accessRequests = await getAccessRequestsByModel(user, model.id)
-    const hasAccessRequest = accessRequests.some((request) =>
-      request.metadata.overview.entities.some((entity) => entities.includes(entity)),
-    )
+  async hasApprovedAccessRequest(user: UserDoc, model: ModelDoc) {
+    const accessRequests = await getModelAccessRequestsForUser(user, model.id)
```

Comment:
I would expect this to be a `service` function, not in a connector.

Also, the implementation is a little odd.  I wouldn't tend to handle the edge case of no access requests.  The speed increase likely isn't worth the complexity?

```typescript
const accessRequests = await getModelAccessRequestsForUser(user, model.id)
return await hasApprovedAccessRequestReview(accessRequests.map((accessRequest) => accessRequest.id))
```

If you did want to keep the check, I'd expect something like:

```typescript
const accessRequests = await getModelAccessRequestsForUser(user, model.id)

if (accessRequests.length === 0) {
  return false
}

const ids = accessRequests.map(request => request.id)
return hasApprovedAccessRequestReview(ids)
```

Reply:
"I would expect this to be a service function, not in a connector." - This was mainly done because the previous implementation made the same decision to make this function in the connector rather than a service. Additionally, it's unclear to me which service this function would belong to as uses both access requests and reviews to make the decision.

The speed increase likely isn't worth the complexity? - The complexity to me seems very minimal considering it's a single if statement to avoid an unnecessary database query, I'd happy to reverse the if statement.